src/libraries/Common/src/System/Text/Json/PooledByteBufferWriter.cs

```diff
@@ -127,6 +127,11 @@ public void Advance(int count)
             Debug.Assert(_index <= _rentedBuffer.Length - count);
 
             _index += count;
+
+            if ((uint)_index > int.MaxValue)
+            {
+                ThrowHelper.ThrowOutOfMemoryException_BufferWriterAdvancedTooFar(_rentedBuffer.Length);
```

Comment:
is this the right exception type to throw? I'd expect InvalidOperationException

Reply:
There seems to be precedent: 

https://github.com/dotnet/runtime/blob/ebdb045532190ffc664bba9a0a1e3f2ce35cf23f/src/libraries/Common/src/System/Text/Json/PooledByteBufferWriter.cs#L216-L222